script.js

```diff
@@ -1,69 +1,101 @@
-function computerPlay() {
-  const choice = ['rock','paper','scissors'];
-  return choice[Math.floor(Math.random()*choice.length)]
+const choices = ['rock', 'paper', 'scissors'];
+
+function randomInteger(max, min = 0) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
 }
 
+function getRandomChoice() {
+  return choices[randomInteger(choices.length) - 1];
+}
 
-function playOneRound(playerSelection, computerSelection) {
+function sanitize(input) {
+  return input.trim().toLowerCase()
+}
 
-  // invalid input
-  if(playerSelection === null || computerSelection === null) {
-    return "Invalid input, try again"
-  }
-  if(playerSelection !== "rock" && playerSelection !== "paper" && playerSelection !== "scissors" ) {
-    return "Invalid input, try again"
+function validate(input) {
+  const errors = [];
+
+  if (!input) {
+    errors.push('Input must not be empty');
   }
 
-  if(playerSelection.toLowerCase() === computerSelection.toLowerCase()) {
-    return "It's a tie!"
+  if (!choices.includes(input)) {
+    errors.push('Input must be one of the following: rock, paper or scissors');
   }
 
-  if (playerSelection.toLowerCase() === 'rock') {
-    if(computerSelection.toLowerCase() === 'scissors') { 
-      return `win! ${playerSelection} beats ${computerSelection}`;
-    }else {
-      return `lost! ${computerSelection} beats ${playerSelection}`;
-    }
+  if (errors.length) {
+    throw new Error(errors.join(', '));
   }
+}
 
-  if (playerSelection.toLowerCase() === 'paper') {
-    if(computerSelection.toLowerCase() === 'rock') { 
-      return `win! ${playerSelection} beats ${computerSelection}`;
-    }else {
-      return `lost! ${computerSelection} beats ${playerSelection}`;
-    }
+function getResult(player1, player2) {
+  const battleMap = {
+    rock: 'scissors',
+    paper: 'rock',
+    scissors: 'paper'
+  };
+
+  if (player1 === player2) {
```

Comment:
Return enum `'win' | 'lose' | 'tie'`. Alternatively `-1 | 0 | 1` but a string might be more explicit